lib/matplotlib/tests/test_backend_pgf.py

```diff
@@ -400,3 +401,24 @@ def test_document_font_size():
              label=r'\normalsize the document font size is \the\fontdimen6\font'
              )
     plt.legend()
+
+
+@needs_pgf_xelatex
+@pytest.mark.backend('pgf')
+def test_pgf_hatch_linewidth():
+    mpl.backend_bases.register_backend('pdf', FigureCanvasPgf)
+    mpl.rcParams['hatch.linewidth'] = 0.1
+
+    plt.bar(1, 1, color='white', edgecolor='black', hatch='/')
+    error = None
+    try:
+        compare_figure('hatch_linewidth.pdf')
+    except ImageComparisonFailure as e:
+        # Store if error
+        error = e
+    # Restore backend to not mess up other tests
+    mpl.backend_bases.register_backend('pdf', FigureCanvasPdf)
```

Comment:
Just put this in a `finally` clause, no need to catch-and-raise and possibly miss other errors.